qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,800 @@
+        for (run, txid) in txids
+        for (to_addr, amount) in run[2]]
+
+def z_mergetoaddress(results, case, zcash, from_addrs, to_addr):
+    return async_txid_cmd(results, case, zcash, 'z_mergetoaddress', [from_addrs, to_addr])
+
+def check_z_mergetoaddress(results, case, zcash, from_addrs, to_addr, amount):
+    txid = z_mergetoaddress(results, case, zcash, from_addrs, to_addr)
+    if txid is None:
+        return decimal.Decimal('0')
+    return wait_and_check_balance(results, case, zcash, to_addr, amount)
+
+def check_z_mergetoaddress_parallel(results, zcash, runs):
+    # First attempt to create all the transactions
+    txids = [(run, z_mergetoaddress(results, run[0], zcash, run[1], run[2])) for run in runs]
+    # Then wait for balance updates caused by successful transactions
+    return [
+        wait_and_check_balance(results, run[0], zcash, run[2], run[3]) if txid is not None else decimal.Decimal('0')
+        for (run, txid) in txids]
+
+
+#
+# Test runners
+#
+
+def simple_commands(zcash):
+    results = {}
+    run_cmd(results, '2a', zcash, 'getinfo'),
+    run_cmd(results, '2b', zcash, 'help'),
+    run_cmd(results, '5o', zcash, 'getwalletinfo'),
+    run_cmd(results, '6a', zcash, 'getpeerinfo'),
+    run_cmd(results, '6b', zcash, 'getnetworkinfo'),
+    run_cmd(results, '6c', zcash, 'getdeprecationinfo'),
+    run_cmd(results, '6d', zcash, 'getconnectioncount'),
+    run_cmd(results, '6e', zcash, 'getaddednodeinfo', [False]),
+    run_cmd(results, '7a', zcash, 'getblocksubsidy'),
+    run_cmd(results, '7c', zcash, 'getmininginfo'),
+    run_cmd(results, '7d', zcash, 'getnetworkhashps'),
+    run_cmd(results, '7e', zcash, 'getnetworksolps'),
+    return results
+
+def transaction_chain(zcash):
+    results = {}
+
+    # Generate the various addresses we will use
+    sprout_zaddr_1 = run_cmd(results, '3a', zcash, 'z_getnewaddress', ['sprout'])
+    sprout_zaddr_2 = run_cmd(results, '3b', zcash, 'z_getnewaddress', ['sprout'])
+    sprout_zaddr_3 = run_cmd(results, '3b', zcash, 'z_getnewaddress', ['sprout'])
+    taddr_1 = run_cmd(results, '3c', zcash, 'getnewaddress')
+    taddr_2 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    taddr_3 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    taddr_4 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    taddr_5 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    sapling_zaddr_1 = run_cmd(results, '3e', zcash, 'z_getnewaddress', ['sapling'])
+    sapling_zaddr_2 = run_cmd(results, '3f', zcash, 'z_getnewaddress', ['sapling'])
+    sapling_zaddr_3 = run_cmd(results, '3f', zcash, 'z_getnewaddress', ['sapling'])
+
+    # Check that the zaddrs are all listed
+    zaddrs = run_cmd(results, '5a', zcash, 'z_listaddresses')
+    if (sprout_zaddr_1 not in zaddrs or
+        sprout_zaddr_2 not in zaddrs or
+        sapling_zaddr_1 not in zaddrs or
+        sapling_zaddr_2 not in zaddrs):
```

Comment:
I'm not super familiar with the python formatting standards, but I get a warning here that seems legit: `visually indented line with same indent as next logical lineflake8(E129)`